### PR TITLE
Add more unit test for DatabaseTypeFactory

### LINKED ISSUE #16293: Add more unit test for DatabaseTypeFactory
Hi community,

We need to add more unit test for DatabaseTypeFactory.

- [ ] DatabaseTypeFactory#getDatabaseType

### PR DESCRIPTION
**Pull request by runqi-zhao**
Fixes #16293.

Changes proposed in this pull request:
- add more unit test for DatabaseTypeFactory

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactory.java [hunk 1]
```diff
@@ -52,4 +59,41 @@ public static DatabaseType getInstance(final String name) {
     public static Collection<DatabaseType> getInstances() {
         return ShardingSphereServiceLoader.getServiceInstances(DatabaseType.class);
     }
+
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactory.java:62]: Why change main codes if the PR is add test cases only?

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactory.java (hunk 1 above)
Comment (runqi-zhao) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactory.java:62]: In issues,  add more unit test for DatabaseTypeFactory.
- DatabaseTypeFactory#getDatabaseType
But I didn't find this method, so I'll define it myself.

Comment (runqi-zhao) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactory.java:62]: I understand. I am learning the code and have finished modifying it

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactory.java [hunk 2]
```diff
@@ -29,7 +29,7 @@
  */
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public final class DatabaseTypeFactory {
-    
+
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactory.java:32]: Please do not change the original indent

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactoryTest.java [hunk 3]
```diff
@@ -57,4 +57,17 @@ public void assertGetInstances() {
         assertThat(iterator.next(), instanceOf(SQLServerDatabaseType.class));
         assertThat(iterator.next(), instanceOf(H2DatabaseType.class));
     }
+    
+    @Test
+    public void assertGetDatabaseType() {
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactoryTest.java:62]: Please rename the method to `assertGetInstance` if the target test method is `getInstance`

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactoryTest.java [hunk 4]
```diff
@@ -57,4 +65,16 @@ public void assertGetInstances() {
...
+    @Test
+    public void assertGetDatabaseTypeWithUrl() {
+        assertThat(DatabaseTypeEngine.getDatabaseType("jdbc:mysql://localhost:3306/test").getType(), is("MySQL"));
+        assertThat(DatabaseTypeEngine.getDatabaseType("jdbc:sqlite:test").getType(), is("SQL92"));
+        assertThat(DatabaseTypeEngine.getDatabaseType("jdbc:postgresql://localhost:5432/test").getType(), is("PostgreSQL"));
+        assertThat(DatabaseTypeEngine.getDatabaseType("jdbc:mariadb://localhost:3306/test").getType(), is("MariaDB"));
+        assertThat(DatabaseTypeEngine.getDatabaseType("jdbc:opengauss://localhost:5432/test").getType(), is("openGauss"));
+        assertThat(DatabaseTypeEngine.getDatabaseType("jdbc:oracle:thin:localhost:1522/test").getType(), is("Oracle"));
+        assertThat(DatabaseTypeEngine.getDatabaseType("jdbc:microsoft:sqlserver://localhost:1433; DatabaseName=test").getType(), is("SQLServer"));
+        assertThat(DatabaseTypeEngine.getDatabaseType("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DATABASE_TO_UPPER=false;MODE=MySQL").getType(), is("H2"));
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeFactoryTest.java:78]: This method tests DatabaseTypeEngine, not DatabaseTypeFactory, and there are duplicated test cases in DatabaseTypeFactory. Can you remove it?

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**